src/targeting/Semgrepignore.ml

```diff
@@ -120,14 +123,24 @@ let create ?include_patterns ?(cli_patterns = []) ~builtin_semgrepignore
   in
   let gitignore_filenames =
     match exclusion_mechanism with
-    | Gitignore_and_semgrepignore ->
-        [ ("gitignore", ".gitignore"); ("semgrepignore", ".semgrepignore") ]
-    | Only_semgrepignore -> [ ("semgrepignore", ".semgrepignore") ]
+    | Gitignore_and_semgrepignore -> [ gitignore_files; semgrep_ignore_files ]
+    | Only_semgrepignore -> [ semgrep_ignore_files ]
+  in
+  (* Check if there is a top level .semgrepignore. If not use builtins *)
+  let semgrep_ignore_exists =
```

Comment:
I'll let @mjambon review this one

Reply:
I didn't realize this was how pysemgrep behaved. This is great.